sympy/concrete/tests/test_gosper.py

```diff
@@ -1,10 +1,11 @@
 """Tests for Gosper's algorithm for hypergeometric summation. """
 
 from sympy import binomial, factorial, gamma, Poly, S, simplify, sqrt, exp, \
-    log, Symbol, pi, Rational
-from sympy.abc import a, b, j, k, m, n, r, x
+    log, Symbol, pi, Rational, symbols
+from sympy.abc import a, b, m, x
 from sympy.concrete.gosper import gosper_normal, gosper_sum, gosper_term
 
+n, k = symbols('n k', nonnegative=True)
```

Comment:
Can we get by with making n vanilla and k nonzero? I would love to see as few changes in this file as possible.

Reply:
Okay, I'll look into that 😄

Reply:
I tried, got a lot of errors popping up 😢 . I've made minimal changes now.

Reply:
The ref [here about Gosper's algorithm](https://mathworld.wolfram.com/GospersAlgorithm.html) talks about `rational functions` which would suggest integer values for k or n - k; it also talks about `nonnegative h`.

Reply:
I just read up on that and a couple other sources. It seems to me that `binomial` with -ve integers isn't exactly a hypergeometric function ?
The hypergeometric function is defined for |z| < 1 by the power series
![](https://wikimedia.org/api/rest_v1/media/math/render/svg/c5689e32546d4e17d239f662b68c397401061e87)
It is undefined (or infinite) if c equals a non-positive integer.